Problem 1

1a

In [ ]:
p = 1

x = 0
for i in range(1, 11):
    x += 1 / (i ** p)
print("Solution for i:", x)

x = 0
for i in range(1, 101):
    x += 1 / (i ** p)
print("Solution for ii:", x)

x = 0
for i in range(1, 10001):
    x += 1 / (i ** p)
print("Solution for iii:", x)


Solution for i: 2.9289682539682538
Solution for ii: 5.187377517639621
Solution for iii: 9.787606036044348


1b

In [ ]:
p = 2
x = 0
for i in range(1, 11):
    x += 1 / (i ** p)
print("Solution for i:", x)

x = 0
for i in range (1,101):
  x += 1/(i**p)
print("solution for ii:", x)

x = 0
for i in range (1,10001):
  x += (1/i**p)
print("solution for iii:", x)

Solution for i: 1.5497677311665408
solution for ii: 1.6349839001848923
solution for iii: 1.6448340718480652


1c

I have't seen any major difffernce caomparing 1a and 1b. In my observation, as the complexity of the calicullations increases the outpit time will increase. In such case, 1b might have a milli seconds difference with 1a.

Problem 2

In [ ]:
vehicles = {"subcompact", "compact", "intermediate", "Luxury", "Truck", "Van"}
total_production = 1.2 * 10**6
compacts_max = 620000
ilx_max = 400000
tv_max = 275000
eco = 27
n_compacts = 0
n_ilx = 0
n_tv = 0
increment = 10
best_profit_so_far = 0
for s in range(1, 601, increment):
    for c in range(1, 401, increment):
        for i in range(1, 301, increment):
            for l in range(1, 226, increment):
                for t in range(1, 326, increment):
                    for v in range(1, 101, increment):
                        if (s + c) < 0.5 * (s + c + l + i):
                            continue
                        if (i + l) * 1000 > ilx_max:
                            continue
                        if (t + v) * 1000 > tv_max:
                            continue
                        if (s + c + i + l + t + v) * 1000 > total_production:
                            continue
                        if (s * 40 + c * 34 + i * 15 + l * 12 + t * 20 + v * 25) / (
                                s + c + i + l + t + v) < eco:
                            continue
                        current_profit = 50 * s + 225 * c + 250 * i + 500 * l + 400 * t + 200 * v
                        if current_profit > best_profit_so_far:
                            best_profit_so_far = current_profit
print("Profit: " + str(best_profit_so_far))


Profit: 334875



1. Optimal profit of ACA is 334875
2. Initially it took longer to execute and the final run time is 8 minutes 51 sec after changing the indentation.

Problem 3

In [ ]:
import scipy.stats as st
import random
import numpy as np

np.random.seed(343)
best_profit_so_far = 0
best_c = 0
best_d = 0
best_t = 0
option_index = 0

random_fabrication_limits = st.norm.rvs(1850, 350, size=10000)
random_assembly_limits = st.norm.rvs(2400, 400, size=10000)
random_shipping_limits = st.norm.rvs(1500, 500, size=10000)

fabrication_quantiles = np.quantile(random_fabrication_limits, 0.1)
assembly_quantiles = np.quantile(random_assembly_limits, 0.1)
shipping_quantiles = np.quantile(random_shipping_limits, 0.1)

for c in range(0, 351, 5):
    for d in range(0, 401, 5):
        for t in range(0, 501, 5):
            x = 4 * c + 6 * d + 2 * t
            if x > 1850:
                continue
            mean_x = np.mean(fabrication_quantiles)  # Expected value
            y = 3 * c + 5 * d + 7 * t
            if y > 2400:
                continue
            mean_y = np.mean(assembly_quantiles)  # Expected value
            z = 3 * c + 2 * d + 4 * t
            if z > 1500:
                continue
            mean_z = np.mean(shipping_quantiles)  # Expected value
            if mean_x * mean_y * mean_z < 0.9:  # Check for probability or confidence levels
                continue
            option_index += 1
            current_solution_profit = 15 * c + 24 * d + 18 * t
            if current_solution_profit > best_profit_so_far:
                best_profit_so_far = current_solution_profit
                best_c = c
                best_d = d
                best_t = t

print(option_index)


100820


In [ ]:
best_c

0

In [ ]:
best_profit_so_far

9000

In [ ]:
best_d

255

In [ ]:
best_t

160

In [ ]:
mean_x

1402.1629014753296

In [ ]:
mean_y

1892.6711192080022

In [ ]:
mean_z

868.6994920584068

Problem 4

If I have to chose one with analytical approach then I will go with the decesion of selling my fathers cable business. It was in 2013 where my father sold out 300+ connection capacity local cable network thinking as there is no scope of future in this business. It was one of our monthly source of incomes and he thought the cable business is going down soon.

I will choose monthly profit as objectives, Opex as variables and capax as constrains. This could have made my family to better decesion of not selling the business. I would have collected the past 10 years od data and projected it with the future demand and profits.